lib/myaccount/request.ts

```diff
@@ -0,0 +1,181 @@
+import { 
+  BaseTransaction,
+  EmailTransaction,
+  EmailStatusTransaction,
+  EmailChallengeTransaction,
+  ProfileTransaction,
+  ProfileSchemaTransaction,
+  PhoneTransaction
+} from './transactions';
+import { httpRequest } from '../http';
+import { AuthApiError, AuthSdkError } from '../errors';
+import { decodeToken, getWithRedirect } from '../oidc';
+import { MyAccountRequestOptions as RequestOptions } from './types';
+import { OktaAuthInterface } from '../types';
+
+export type TransactionLink = {
+  href: string;
+  hints?: {
+    allow?: string[];
+  };
+}
+
+type TransactionLinks = {
+  self: TransactionLink;
+  [property: string]: TransactionLink;
+}
+
+type SendRequestOptions = RequestOptions & {
+  url: string;
+  method: string;
+  transactionClassName?: string;
+}
+
+type InsufficientAuthenticationError = {
+  error: string;
+  // eslint-disable-next-line camelcase
+  error_description: string;
+  // eslint-disable-next-line camelcase
+  max_age: string;
+}
+
+const parseInsufficientAuthenticationError = (
+  header: string
+): InsufficientAuthenticationError => {
+  if (!header) {
+    throw new AuthSdkError('Missing header string');
+  }
+
+  return header
+    .split(',')
+    .map(part => part.trim())
+    .map(part => part.split('='))
+    .reduce((acc, curr) => {
+      acc[curr[0]] = curr[1];
+      return acc;
+    }, {}) as InsufficientAuthenticationError;
+};
+
+/* eslint-disable complexity */
+export async function sendRequest<T extends BaseTransaction> (
+  oktaAuth: OktaAuthInterface, 
+  options: SendRequestOptions
+): Promise<T | T[]> {
+  const { 
+    accessToken: accessTokenObj,
+    idToken: idTokenObj 
+  } = oktaAuth.tokenManager.getTokensSync();
+  
+  const idToken = idTokenObj?.idToken;
+  const accessToken = options.accessToken || accessTokenObj?.accessToken;
+  const { issuer } = oktaAuth.options;
+  const { url, method, payload } = options;
+  const requestUrl = url.startsWith(issuer!) ? url : `${issuer}${url}`;
+
+  if (!accessToken) {
+    throw new AuthSdkError('AccessToken is required to request MyAccount API endpoints.');
+  }
+  
+  let res;
+  try {
+    res = await httpRequest(oktaAuth, {
+      headers: { 'Accept': '*/*;okta-version=1.0.0' },
+      accessToken,
+      url: requestUrl,
+      method,
+      ...(payload && { args: payload })
+    });
+  } catch (err) {
+    const errorResp = (err as AuthApiError).xhr;
+    if (idToken && errorResp?.status === 403 && !!errorResp?.headers?.['www-authenticate']) {
+      const error = parseInsufficientAuthenticationError(errorResp?.headers?.['www-authenticate']);
+      if (error?.error?.includes('insufficient_authentication_context')) {
+        const scopes = decodeToken(accessToken).payload.scp!;
+        // reauthentication - this call triggers redirect
+        await getWithRedirect(
```

Comment:
This seems incorrect. According to this JIRA - https://oktainc.atlassian.net/browse/OKTA-460659, 
>When a response includes insufficient_authentication_context that should trigger a new interact/authorize request including the max_age parameter as defined in the response
In our case, we need to call `interact` again instead of the `getWithRedirect`
The current experience redirects the user to okta hosted page which is undesired behavior.